core/modules/qana/PostPlugin.cc

```diff
@@ -128,49 +129,46 @@ PostPlugin::applyLogical(query::SelectStmt& stmt,
 }
 
 void
-PostPlugin::applyPhysical(QueryPlugin::Plan& p,
+PostPlugin::applyPhysical(QueryPlugin::Plan& plan,
                           query::QueryContext& context) {
     // Idea: If a limit is available in the user query, compose a
-    // merge statement (if one is not available) and turn on merge
-    // fixup.
+    // merge statement (if one is not available)
     LOGF(_logger, LOG_LVL_DEBUG, "Apply physical");
-    if(context.hasChunks()) { // For chunked queries only.
-        if((_limit != -1) || _orderBy) { // Aggregating LIMIT or ORDER BY
-            // Prepare merge statment.
-            // If empty select in merger, create one with *
-            query::SelectList& mList = p.stmtMerge.getSelectList();
-            std::shared_ptr<query::ValueExprPtrVector> vlist;
-            vlist = mList.getValueExprList();
-            if(!vlist) {
-                throw std::logic_error("Unexpected NULL ValueExpr in SelectList");
-            }
-            if(vlist->size() == 0) {
-                mList.addStar(std::string());
-            }
-            // Patch MergeFixup.
-            context.needsMerge = true;
-            LOGF(_logger, LOG_LVL_DEBUG, "Add merge operation");
-
-            if(_orderBy) {
-                // Remove orderby from parallel
-                // (no need to sort until we have all the results)
-                SelectStmtPtrVector::iterator i, e;
-                std::shared_ptr<query::OrderByClause> _nullptr;
-                for(i=p.stmtParallel.begin(), e=p.stmtParallel.end(); i != e; ++i) {
-                    (**i).setOrderBy(_nullptr);
-                }
-                // Make sure the merge has an ORDER BY
-                LOGF(_logger, LOG_LVL_DEBUG, "Add ORDER BY clause %1%" % _orderBy);
-                p.stmtMerge.setOrderBy(_orderBy);
-            }
+
+
+    if (_limit!=NOTSET) {
+        // [ORDER BY ...] LIMIT ... is a special case which require sort on worker and sort/aggregation on czar
+        if (context.hasChunks()) {
+             LOGF(_logger, LOG_LVL_DEBUG, "Add merge operation");
+             context.needsMerge = true;
+         }
+    }
+    else if (_orderBy) {
+        // If there is no LIMIT clause, remove ORDER BY clause from all Czar queries because it is performed by
+        // mysql-proxy (mysql doesn't garantee result order for non ORDER BY queries)
+        std::shared_ptr<query::OrderByClause> _nullptr;
```

Comment:
underscore is used for private members, regular variable names start with letter.